### DISCUSSION TITLE
[BUG] Not reading `flake8` config from `tox.ini`

### DISCUSSION THREAD

**Discussion by Peque** [s-weigand/flake8-nb#245] [Q&A] [answered]
**Describe the bug**

I would expect `flake8-nb` to honor the `flake8` configuration set in `tox.ini`.

When having a mix of Python (`.py`) and Notebook (`.ipynb`) files in a project/repo, it is nice to use the same style/rules in all the code. :blush:

As an examle, `black[jupyter]` honors the default Black configuration file if it is present.

**To Reproduce**

Write a `tox.ini` file with this content:

```
[flake8]
max-line-length = 20
```

Then write this code in a notebook cell:

```
a_long_line = "............................"
```

Run `flake8-nb` and see no error.

**Expected behavior**

```
E501 line too long (44 > 20 characters)
```

**Comment by s-weigand** [comment, marked as answer]
If you want to achieve this you need to slightly [change your configuration](https://flake8-nb.readthedocs.io/en/latest/usage.html#project-wide-configuration) and use 
```ini
[flake8_nb]
max-line-length = 20
```

The reason for this is that `flake8-nb` hacks the internals of `flake8` to treat the Notebooks like a Schödingers cat where they are ignored (json blob, which a notebook is under the hood) and checked (converted to python file version which gets mapped back in the reporter).

The origin of this project and what it is mostly geared to is writing teaching material and tutorials, where you want to keep up code quality, but at the same time demonstrate "bad code" without adding the complexity of linter exception ([cell tag configuration](https://flake8-nb.readthedocs.io/en/latest/usage.html#cell-tags)). That said this is more a feature than a bug, since it also allows to define different rules for notebooks and python files.
E.g.:

```ini
[flake8_nb]
filename = *.ipynb
extend-ignore = 
    # All module level imports should be at the top of the file. 
    E402
```

If you are interested in a more production-oriented solution (all the code passes `flake8` with the normal config, which you can just copy-paste across projects) [nbQA](https://github.com/nbQA-dev/nbQA) might be the better suited for your project. `nbQA` respects the normal configurations since it first parses the notebooks and after that hands them over to the tool you want to use.

**Comment by Peque** [comment]
@s-weigand Thanks for the clarification! :blush:
